framework/test_app/runners/k8s/k8s_base_runner.py

```diff
@@ -299,6 +299,34 @@ def _reuse_namespace(self) -> k8s.V1Namespace:
         logger.info("Reusing namespace: %s", self.k8s_namespace.name)
         return self.k8s_namespace.get()
 
+    def _create_pod_monitoring(self, template, **kwargs) -> None:
+        if not kwargs["namespace_name"]:
```

Comment:
Ok, so I double checked and this does not work as intended. 

```py
    303 def _create_pod_monitoring_stk(cls, template, **kwargs) -> None:
--> 304     if not kwargs["namespace_name"]:
    305         raise _RunnerError(
    306             "namespace_name required to create PodMonitoring resource"
    307         )
    308     if not kwargs["deployment_id"]:

KeyError: 'namespace_name'
```

However, I don't think we want to check args in the first place. See my suggestion below.

Reply:
Fixed